gui/app/simple_clock.h

```diff
@@ -0,0 +1,62 @@
+#ifndef CLOCK_H_
+#define CLOCK_H_
+
+#include <QObject>
+#include <QTimer>
+
+/**
+ * @brief Clock class that emit current datetime in a
+ * given interval
+ */
+class SimpleClock : public QObject {
+  Q_OBJECT
+  Q_PROPERTY(
+      bool enabled READ GetEnabled WRITE SetEnabled NOTIFY enabledChanged)
+  Q_PROPERTY(int hour READ GetHour NOTIFY hourChanged)
+  Q_PROPERTY(int minute READ GetMinute NOTIFY minuteChanged)
+  Q_PROPERTY(int second READ GetSecond NOTIFY secondChanged)
+
+public:
+  explicit SimpleClock(QObject *parent = nullptr) : QObject(parent) {
+    connect(&timer_, &QTimer::timeout, this, &SimpleClock::update_timer);
+    timer_.start();
+  }
+
+  ~SimpleClock() { timer_.stop(); }
+
+  bool GetEnabled() const { return enabled_; }
+
+  int GetUpdateInterval() const { return update_interval_; }
+
+  int GetHour() const { return hour_; }
+  int GetMinute() const { return minute_; }
+  int GetSecond() const { return second_; }
+
+public slots:
+  void SetEnabled(bool enabled) {
+    if (enabled_ != enabled) {
```

Comment:
Can't reply to the other comment for some reason, but yeah, I would prefer to remove this unnecessary functionality of changing enabled and update interval. It is "dead-on-arrival" code, it's best not to add such code.

Reply:
I just removed it